cpp/include/raft/core/mdspan.hpp

```diff
@@ -32,6 +32,67 @@ template <typename ElementType,
           typename AccessorPolicy = std::experimental::default_accessor<ElementType>>
 using mdspan = std::experimental::mdspan<ElementType, Extents, LayoutPolicy, AccessorPolicy>;
 
+/**
+ * Some helper templates to handle mdspan with padded layouts / aligned memory
+ */
+enum class StorageOrderType { column_major_t, row_major_t };
+
+namespace detail {
```

Comment:
Can we stick this in raft/core/detail/mdspan.hpp just to keep the public API clean?

Reply:
P2642 has moved away from using tag types to express storage order.  It provides separate types `layout_left_padded` and `layout_right_padded`, for consistency with the Standard layouts `layout_left` and `layout_right`.

Reply:
I have removed the enum and provided two separate layouts.